docs/ref/models/expressions.txt

```diff
@@ -450,6 +450,76 @@ Conditional expressions allow you to use :keyword:`if` ... :keyword:`elif` ...
 :keyword:`else` logic in queries. Django natively supports SQL ``CASE``
 expressions. For more details see :doc:`conditional-expressions`.
 
+``SubQuery`` expressions
+------------------------
+
+.. class:: SubQuery(queryset, output_field=None)
+
+.. versionadded:: 1.11
+
+When a value is required that is dependent upon values from within a related
+object, it's possible to use an explicit ``SubQuery``. For example::
+
+    >>> largest = Size.objects.filter(relation=OuterRef('pk')).order_by('-size').values('size')
+    >>> queryset.annotate(largest_size=SubQuery(largest[:1]))
+
+.. note::
+
+  The inner queryset contains an ``OuterRef`` expression, which refers to a field from the
+  outer queryset. This acts like an ``F`` expression, but the check on if it is valid field
+  name is not made until the outer queryset is "resolved".
+
+.. note::
+
+  Something about how ``.values('size')`` is used, and why it is required in this case.
+
+.. note::
+
+  To prevent multiple rows being returned in a subquery, a slice (``[:1]``) of the queryset is used,
+  rather than a `.get()`, which would force evaluation of the queryset at that time. That
+  would fail, because at that time, the ``OuterRef`` would not be able to be resolved.
+
+.. note::
+
+  It is possible to nest ``SubQuery`` expressions, and refer to an outer query other than
+  the immediate parent by nesting calls to ``SubQuery``::
+
+      >>> queryset = Size.objects.filter(grandparent=OuterRef(OuterRef('pk')))
+
+.. class:: Exists(queryset)
+
+.. versionadded:: 1.11
+
+Based upon the ``SubQuery`` expression is the ``Exists`` expression, that
+generates an efficient subquery in the case of a boolean test:
+
+    >>> subquery = Child.objects.filter(parent=OuterRef('pk), date_range__contains=now())
+    >>> queryset.annotate(is_valid=Exists(subquery))
+
+.. note::
+
+  It is not necessary to force an ``Exists`` to refer to a single column, as the object will
+  perform that operation automatically.
+
+.. warning::
+
+  It is not currently possible to filter directly using an `Exists`:
+
+      >>> queryset.filter(Exists(subquery))
+
+You may filter on a subquery expression by first annotating the queryset, and then filtering
+based on that annotation:
+
+    >>> queryset.annotate(is_valid=Exists(subquery)).filter(is_valid=True)
+    >>> queryset.annotate(largest=SubQuery(subquery)).filter(largest__gte=55)
+
+.. warning::
+
+  It is not currently possible to use an aggregation within a SubQuery:
+
+      >>> inner = Child.objects.filter(relation=OuterRef('pk'))
+      >>> queryset.filter(length__lte=SubQuery(inner.annotate(the_sum=Sum('target')).values('the_sum')))
```

Comment:
I did have a guy in IRC discussing issues which I think were related to aggregations within a subquery. I asked him to comment here, I'll try chase up what was not working there.